hazelcast/src/main/java/com/hazelcast/instance/HazelcastInstanceProxy.java

```diff
@@ -68,7 +68,7 @@
  * </li>
  * </ol>
  */
-public final class HazelcastInstanceProxy implements HazelcastInstance, SerializationServiceSupport {
+public class HazelcastInstanceProxy implements HazelcastInstance, SerializationServiceSupport {
     protected volatile HazelcastInstanceImpl original;
```

Comment:
Moving to internal could lead to problems with e.g. simulator since we rely on this class for reflection.

I would add a @PrivateAPI annotation on top of the class. That should be sufficient. This instance package is a mixture of public and private classes.